tests/ga/test_extension.py

```diff
@@ -1513,26 +1475,129 @@ def test_upgrade_failure_with_exception_handling(self, patch_get_update_command,
+    def test_extension_upgrade_should_fail_when_continue_on_update_failue_is_true_and_old_disable_and_new_enable_fails(
+            self, patch_get_disable_command, *args):
+        test_data, exthandlers_handler, protocol = self._setup_update_test_and_update_gs(patch_get_disable_command,
+                                                                                         *args)
+
+        with patch('azurelinuxagent.ga.exthandlers.HandlerManifest.is_continue_on_update_failure', return_value=True) \
+                as mock_continue_on_update_failure:
+            with patch('azurelinuxagent.ga.exthandlers.HandlerManifest.get_enable_command', return_value="exit 1")\
+                    as patch_get_enable:
+                exthandlers_handler.run()
+                self.assertEqual(1, patch_get_disable_command.call_count)
+                self.assertEqual(2, mock_continue_on_update_failure.call_count)
+                self.assertEqual(1, patch_get_enable.call_count)
+
+        self._assert_handler_status(protocol.report_vm_status, "NotReady", expected_ext_count=1, version="1.0.1")
+
+    @patch('azurelinuxagent.ga.exthandlers.HandlerManifest.is_continue_on_update_failure', return_value=True)
+    @patch('azurelinuxagent.ga.exthandlers.HandlerManifest.get_uninstall_command')
+    def test_both_env_var_should_clear_after_last_enable_call_on_failure_when_continue_on_update_failue_is_true(
+            self, patch_get_uninstall_command, patch_continue_on_update, *args):
+        test_data, exthandlers_handler, protocol = self._setup_update_test_and_update_gs(patch_get_uninstall_command,
+                                                                                         *args)
+
+        with patch.object(ExtHandlerInstance, 'remove_key_from_handler_env', autospec=True) as patch_remove_env_variable:
+            exthandlers_handler.run()
+            disable_args, disable_kwargs = patch_remove_env_variable.call_args_list[0]
+            uninstall_args, uninstall_kwargs = patch_remove_env_variable.call_args_list[1]
+
+            self.assertEqual(1, patch_get_uninstall_command.call_count)
+            self.assertEqual(2, patch_continue_on_update.call_count)
+            self.assertEqual(2, patch_remove_env_variable.call_count,
+                             "This should be called twice, Once after update and once after enable")
+            self.assertTrue(DISABLE_FAILED == disable_args[1])
+            self.assertTrue(UNINSTALL_FAILED == uninstall_args[1])
+
+        self._assert_handler_status(protocol.report_vm_status, "Ready", expected_ext_count=1, version="1.0.1")
+
+    def test_ext_path_and_version_env_variables_set_for_ever_operation(self, *args):
+        test_data = WireProtocolData(DATA_FILE_EXT_SINGLE)
+        exthandlers_handler, protocol = self._create_mock(test_data, *args)
+
+        with patch('azurelinuxagent.ga.exthandlers.ExtHandlerInstance._capture_process_output',
```

Comment:
Depending on which PR goes in first, this logic changes with #1623

Reply:
Ohh yeah. Your changes would actually make that patch obsolete and better :)
The fix would be super simple for this test case whenever we need to change it